src/main/java/org/reaktivity/nukleus/socks/internal/stream/client/ConnectReplyStreamProcessor.java

```diff
@@ -0,0 +1,422 @@
+import org.reaktivity.nukleus.socks.internal.types.stream.WindowFW;
+
+final class ConnectReplyStreamProcessor extends AbstractStreamProcessor
+{
+
+    private MessageConsumer streamState;
+
+    private int acceptReplyWindowBytesAdjustment;
+    private int acceptReplyWindowFramesAdjustment;
+
+    private int slotIndex = NO_SLOT;
+    private int slotOffset;
+    Correlation correlation;
+    private final MessageConsumer connectReplyThrottle;
+    private final long connectReplyStreamId;
+
+    ConnectReplyStreamProcessor(
+        Context context,
+        MessageConsumer connectReplyThrottle,
+        long connectReplyId)
+    {
+        super(context);
+        this.streamState = this::beforeBegin;
+        this.connectReplyThrottle = connectReplyThrottle;
+        this.connectReplyStreamId = connectReplyId;
+    }
+
+    @Override
+    protected void handleStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        streamState.accept(msgTypeId, buffer, index, length);
+    }
+
+    @State
+    private void beforeBegin(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        if (msgTypeId == BeginFW.TYPE_ID)
+        {
+            final BeginFW begin = context.beginRO.wrap(buffer, index, index + length);
+            handleBegin(begin);
+        }
+        else
+        {
+            doReset(connectReplyThrottle, connectReplyStreamId);
+        }
+    }
+
+    @State
+    private void beforeNegotiationResponse(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch (msgTypeId)
+        {
+        case DataFW.TYPE_ID:
+            final DataFW data = context.dataRO.wrap(buffer, index, index + length);
+            handleNegotiationResponse(data);
+            break;
+        case EndFW.TYPE_ID:
+            final EndFW end = context.endRO.wrap(buffer, index, index + length);
+            handleEnd(end);
+            break;
+        case AbortFW.TYPE_ID:
+            final AbortFW abort = context.abortRO.wrap(buffer, index, index + length);
+            handleAbort(abort);
+            break;
+        default:
+            doReset(connectReplyThrottle, connectReplyStreamId);
+            break;
+        }
+    }
+
+    private void handleBegin(BeginFW begin)
+    {
+        final long connectRef = begin.sourceRef();
+        final long correlationId = begin.correlationId();
+        correlation = context.correlations.remove(correlationId);
+        if (connectRef == 0L && correlation != null)
+        {
+            context.router.setThrottle(
+                correlation.acceptSourceName(),
+                correlation.acceptReplyStreamId(),
+                this::handleAcceptReplyThrottle);
+            final long acceptReplyRef = 0; // Bi-directional reply
+            BeginFW beginToAcceptReply = context.beginRW
```

Comment:
consider moving the begins and all writes to a writer class so updates to extension.reset etc are all in the same place